Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py

```diff
@@ -212,6 +213,45 @@ def populate(self, issue, member=None):
 
         return issue
 
+    def set(self, issue, assignee=None, opened=None, why=None, **properties):
+        update_dict = dict()
+
+        if assignee:
+            if not isinstance(assignee, User):
+                raise TypeError("Must assign to '{}', not '{}'".format(User, type(assignee)))
+            issue._assignee = self.user(name=assignee.name, username=assignee.username, email=assignee.email)
+            update_dict['assigned_to'] = issue._assignee.username
+
+        if opened is not None:
+            issue._opened = bool(opened)
+            if issue._opened:
+                update_dict['status'] = 'REOPENED'
+                why = why or 'Reopening bug'
+            else:
+                update_dict['status'] = 'RESOLVED'
+                update_dict['resolution'] = 'FIXED'
+
+        if why is not None:
+            update_dict['comment'] = dict(body=why)
+
+        if properties:
+            raise TypeError("'{}' is an invalid property".format(list(properties.keys())[0]))
```

Comment:
Should we move this to be the first check so that we can early return if any (unsupported) properties is specified?

Reply:
Good point! Originally intended to do `properties.pop`, but named arguments ends up being cleaner.